stdlib/public/core/Sequence.swift

```diff
@@ -877,6 +877,13 @@ extension Sequence {
   public func filter(
     _ isIncluded: (Element) throws -> Bool
   ) rethrows -> [Element] {
+    return try _filter(isIncluded)
```

Comment:
Oh, I see what you mean here... The line you commented on is misleading, but yeah, even with the addition of a new overload, they will still return `[T]`.